test/elixir/test/jwtauth_test.exs

```diff
@@ -16,17 +16,43 @@ defmodule JwtAuthTest do
     ]
 
     run_on_modified_server(server_config, fn ->
-      test_fun()
+      resp = Couch.get("/_session",
+        headers: [authorization: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJjb3VjaEBhcGFjaGUub3JnIiwicm9sZXMiOlsiX2FkbWluIl19.JF_vunmn95R-XxQK1UtYhlKdLJdg4ms3sBeZRxpXgkw"]
+      )
+
+      assert resp.body["userCtx"]["name"] == "couch@apache.org"
+      assert resp.body["userCtx"]["roles"] == [<<"_admin">>]
+      assert resp.body["info"]["authenticated"] == "jwt"
+    end)
+  end
+
+  test "jwt auth with secret and without roles", _context do
+
+    secret = "zxczxc12zxczxc12"
+
+    server_config = [
+      %{
+        :section => "jwt_auth",
+        :key => "secret",
+        :value => secret
+      }
+    ]
+
+    run_on_modified_server(server_config, fn ->
+      run_on_modified_server(server_config, fn ->
+        resp = Couch.get("/_session",
+          headers: [authorization: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJjb3VjaEBhcGFjaGUub3JnIn0.KYHmGXWj0HNHzZCjfOfsIfZWdguEBSn31jUdDUA9118"]
```

Comment:
see in my open PR how to encode the token directly (https://github.com/apache/couchdb/pull/2687/files#diff-b72decaf14c27ef5e205025f5fa0c011R106). that way the test is clearer to the reader.